src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts

```diff
@@ -9,16 +9,46 @@ import { Alarm } from 'aws-cdk-lib/aws-cloudwatch';
 import { Cluster, IService } from 'aws-cdk-lib/aws-ecs';
 import { Rule, Schedule } from 'aws-cdk-lib/aws-events';
 import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
-import { IRole } from 'aws-cdk-lib/aws-iam';
+import { Effect, IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
 import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
 import { Construct } from 'constructs';
 
 export interface EcsIsoServiceAutoscalerProps {
   /**
-   * The IAM role that allows the created lambda to adjust the desired count on the ECS Service .
-   * TODO: Make optional and add function for auto generation
+   * Optional IAM role to attach to the created lambda to adjust the desired count on the ECS Service
+   *
+   * Ensure this role has appropriate privileges. Example IAM policy statements:
+   * And more
```

Comment:
This seems out of place.

Reply:
The idea is that if a developer needs for a separate team in the enterprise to create a role on their behalf to make sure it has sufficient privileges to do it's autoscaling job